From 61cde69b8f21c42e64c1ae8236a0517055763f44 Mon Sep 17 00:00:00 2001
From: Matthew Penner <me@matthewp.io>
Date: Wed, 8 Apr 2026 10:48:52 -0600
Subject: [PATCH 2/4] libblake3: ensure `useTBB` remains disabled

Signed-off-by: Matthew Penner <me@matthewp.io>
---
 packaging/dependencies.nix | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/packaging/dependencies.nix b/packaging/dependencies.nix
index f5e39e6e0..5a5c3400b 100644
--- a/packaging/dependencies.nix
+++ b/packaging/dependencies.nix
@@ -17,11 +17,11 @@ scope: {
   inherit stdenv;
 
   libblake3 =
-    (pkgs.libblake3.override {
+    (pkgs.libblake3.override (oldAttrs: {
       inherit stdenv;
       # Nixpkgs disables tbb on static
-      useTBB = !stdenv.hostPlatform.isStatic;
-    })
+      useTBB = oldAttrs.useTBB && !stdenv.hostPlatform.isStatic;
+    }))
     # For some reason that is not clear, it is wanting to use libgcc_eh which is not available.
     # Force this to be built with compiler-rt & libunwind over libgcc_eh works.
     # Issue: https://github.com/NixOS/nixpkgs/issues/177129
-- 
2.53.0

