From 999814e6c130f9f69536cc366532cc94643c864f Mon Sep 17 00:00:00 2001
From: Matthew Penner <me@matthewp.io>
Date: Wed, 8 Apr 2026 11:06:47 -0600
Subject: [PATCH 4/4] lowdown: restrict to `2.x.x` versions only

Signed-off-by: Matthew Penner <me@matthewp.io>
---
 packaging/dependencies.nix | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/packaging/dependencies.nix b/packaging/dependencies.nix
index 5a5c3400b..0554be91b 100644
--- a/packaging/dependencies.nix
+++ b/packaging/dependencies.nix
@@ -89,20 +89,25 @@ scope: {
       });
 
   lowdown =
-    if lib.versionAtLeast pkgs.lowdown.version "2.0.2" then
+    if (lib.versions.major pkgs.lowdown.version == "2" && lib.versionAtLeast pkgs.lowdown.version "2.0.2") then
       pkgs.lowdown
     else
       pkgs.lowdown.overrideAttrs (prevAttrs: rec {
-        version = "2.0.2";
+        version = "2.0.4";
         src = pkgs.fetchurl {
           url = "https://kristaps.bsd.lv/lowdown/snapshots/lowdown-${version}.tar.gz";
-          hash = "sha512-cfzhuF4EnGmLJf5EGSIbWqJItY3npbRSALm+GarZ7SMU7Hr1xw0gtBFMpOdi5PBar4TgtvbnG4oRPh+COINGlA==";
+          hash = "sha512-ZJpQi3cn325+EgOrs4U+BfFntkgy/V4ScfFCzPeC5gCx3nPHLcAmc9exde/9xU8sD2AxggipaK+fl2PQnPT57w==";
         };
         patches = [ ];
         nativeBuildInputs = prevAttrs.nativeBuildInputs ++ [ pkgs.buildPackages.bmake ];
         postInstall =
           lib.replaceStrings [ "lowdown.so.1" "lowdown.1.dylib" ] [ "lowdown.so.2" "lowdown.2.dylib" ]
             (prevAttrs.postInstall or "");
+        postPatch = ''
+          # fails test, some column width mismatch
+          rm regress/table-footnotes.md
+          rm regress/table-styles.md
+        '';
       });
 
   # TODO: Remove this when https://github.com/NixOS/nixpkgs/pull/442682 is included in a stable release
-- 
2.53.0

