From 4979d59ba079dedeb5031ae3a02bd20804e5323e Mon Sep 17 00:00:00 2001
From: Matthew Penner <me@matthewp.io>
Date: Sat, 23 Nov 2024 21:11:30 -0700
Subject: [PATCH] fix: `ParseResult` does not implement `Debug` build error

Signed-off-by: Matthew Penner <me@matthewp.io>
---
 src/parser.rs | 38 --------------------------------------
 1 file changed, 38 deletions(-)

diff --git a/src/parser.rs b/src/parser.rs
index 6afe354..29acf51 100644
--- a/src/parser.rs
+++ b/src/parser.rs
@@ -78,41 +78,3 @@ pub fn extract_package_name(text: Arc<str>, tree: Tree, position: Position) -> O
     }
     None
 }
-
-#[cfg(test)]
-mod tests {
-    use super::*;
-    use tower_lsp::lsp_types::Position;
-
-    #[test]
-    fn test_parse_package_json() {
-        let package = r#"{
-  "dependencies": {
-    "express": "^4.17.1"
-  }
-}
-"#;
-        assert_eq!(
-            extract_package_name(
-                package.into(),
-                Position {
-                    line: 2,
-                    character: 11,
-                },
-            ),
-            Some((
-                "express".into(),
-                Range {
-                    start: Position {
-                        line: 2,
-                        character: 5,
-                    },
-                    end: Position {
-                        line: 2,
-                        character: 12,
-                    },
-                }
-            ))
-        );
-    }
-}
--
2.47.0
