From a0a49e05e655f46e8858a93aae8a5f2949937c07 Mon Sep 17 00:00:00 2001
From: Matthew Penner <me@matthewp.io>
Date: Sun, 8 Dec 2024 13:33:56 -0700
Subject: [PATCH 2/5] chore: enable electron options for better hardware
 acceleration

Signed-off-by: Matthew Penner <me@matthewp.io>
---
 src/main/index.ts | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/src/main/index.ts b/src/main/index.ts
index 3d74a55..f6de7e6 100644
--- a/src/main/index.ts
+++ b/src/main/index.ts
@@ -48,6 +48,23 @@ function init() {
     app.commandLine.appendSwitch("disable-renderer-backgrounding");
     app.commandLine.appendSwitch("disable-background-timer-throttling");
     app.commandLine.appendSwitch("disable-backgrounding-occluded-windows");
+    app.commandLine.appendSwitch("enable-zero-copy");
+    app.commandLine.appendSwitch("use-gl=angle");
+    app.commandLine.appendSwitch("use-angle=gl");
+    app.commandLine.appendSwitch("use-vulkan");
+    app.commandLine.appendSwitch("enable-oop-rasterization");
+    app.commandLine.appendSwitch("enable-raw-draw");
+    app.commandLine.appendSwitch("enable-gpu-rasterization");
+    app.commandLine.appendSwitch("enable-gpu-compositing");
+    app.commandLine.appendSwitch("enable-native-gpu-memory-buffers");
+    app.commandLine.appendSwitch("enable-accelerated-2d-canvas");
+    app.commandLine.appendSwitch("enable-accelerated-video-decode");
+    app.commandLine.appendSwitch("enable-accelerated-mjpeg-decode");
+    app.commandLine.appendSwitch("disable-gpu-vsync");
+    app.commandLine.appendSwitch("disable-frame-rate-limit");
+    app.commandLine.appendSwitch("ozone-platform-hint=auto");
+    app.commandLine.appendSwitch("enable-webrtc-pipewire-capturer");
+    // app.commandLine.appendSwitch("ozone-platform=wayland");
     if (process.platform === "win32") {
         disabledFeatures.push("CalculateNativeWinOcclusion");
     }
--
2.47.0
