From cbcefbe7528b492180bd1faa8849c27f804b6c4f Mon Sep 17 00:00:00 2001
From: Matthew Penner <me@matthewp.io>
Date: Sun, 8 Dec 2024 13:34:09 -0700
Subject: [PATCH 3/5] chore: increase screensharing bitrate

Signed-off-by: Matthew Penner <me@matthewp.io>
---
 src/renderer/components/ScreenSharePicker.tsx | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/renderer/components/ScreenSharePicker.tsx b/src/renderer/components/ScreenSharePicker.tsx
index 2055067..328ae8a 100644
--- a/src/renderer/components/ScreenSharePicker.tsx
+++ b/src/renderer/components/ScreenSharePicker.tsx
@@ -84,9 +84,9 @@ addPatch({
         const width = Math.round(height * (16 / 9));

         Object.assign(opts, {
-            bitrateMin: 500000,
-            bitrateMax: 8000000,
-            bitrateTarget: 600000
+            bitrateMin: 10000000,
+            bitrateMax: 60000000,
+            bitrateTarget: 32000000
         });
         if (opts?.encode) {
             Object.assign(opts.encode, {
--
2.47.0
