From ef68253f88a7526b5cd2403ffe9a0f0083d12732 Mon Sep 17 00:00:00 2001
From: Matthew Penner <me@matthewp.io>
Date: Wed, 29 Jan 2025 11:06:24 -0700
Subject: [PATCH] Enable `package-version-server` lookup in PATH

Signed-off-by: Matthew Penner <me@matthewp.io>
---
 crates/languages/src/json.rs | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/crates/languages/src/json.rs b/crates/languages/src/json.rs
index 95aefd405a..78fbe55a5e 100644
--- a/crates/languages/src/json.rs
+++ b/crates/languages/src/json.rs
@@ -335,6 +335,21 @@ impl LspAdapter for NodeVersionAdapter {
         }))
     }
 
+    async fn check_if_user_installed(
+        &self,
+        delegate: &dyn LspAdapterDelegate,
+        _: Arc<dyn LanguageToolchainStore>,
+        _: &AsyncAppContext,
+    ) -> Option<LanguageServerBinary> {
+        let path = delegate.which("package-version-server".as_ref()).await?;
+
+        Some(LanguageServerBinary {
+            path,
+            env: None,
+            arguments: Default::default(),
+        })
+    }
+
     async fn fetch_server_binary(
         &self,
         latest_version: Box<dyn 'static + Send + Any>,
-- 
2.47.1

